### PR TITLE
KYLIN-3997: Add a health check job of Kylin

### PR DESCRIPTION
**Pull request by lshmouse**
See: https://issues.apache.org/jira/browse/KYLIN-3997

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: server-base/src/main/java/org/apache/kylin/rest/job/KylinHealthCheckJob.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
+/*
```
Comment (hit-lacus) [inline server-base/src/main/java/org/apache/kylin/rest/job/KylinHealthCheckJob.java]: Hi, @lshmouse , I found this java file is not formatted as expected. It looks like it is indented with two blank and it is different with other Java file, maybe you should check **Setup IDE code formatter** in http://kylin.apache.org/development/dev_env.html.

File: server-base/src/main/java/org/apache/kylin/rest/job/KylinHealthCheckJob.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    DataModelManager modelManager = DataModelManager.getInstance(config);
+    for (String model: modelManager.getErrorModels()) {
+      reporter.log("Error loading DataModelDesc at " + model);
+    }
+  }
+
+  private void checkStoppedJob() throws Exception {
+    reporter.log("## Cleanup stopped job");
+    // TODO: Add a config
+    long outdatedJobTimeCut = System.currentTimeMillis() - 30 * 24 * 3600 * 1000L;
```
Comment (hit-lacus) [inline server-base/src/main/java/org/apache/kylin/rest/job/KylinHealthCheckJob.java]: It looks like that `30 * 24 * 3600 * 1000L` is a magic number, could you please add these in **KylinConfig**?

File: core-cube/src/main/java/org/apache/kylin/cube/CubeDescManager.java [hunk 3]
```diff
@@ -164,6 +164,10 @@ public CubeDesc reloadCubeDescQuietly(String name) {
         }
     }
 
+    public List<String> getErrorCubes() {
```
Comment (hit-lacus) [inline core-cube/src/main/java/org/apache/kylin/cube/CubeDescManager.java]: This method is duplicated with `getErrorCubes` in `CubeManager.java`

Comment (hit-lacus) [reply core-cube/src/main/java/org/apache/kylin/cube/CubeDescManager.java]: And it is never called anywhere.

**Comment by hit-lacus** [comment]
I think it is a good feature, by this feature, Kylin's administrator needn't check metadata error in kylin.log or find useless cube manually. Some value such as *ExpansionRate* or *Out of date* should be configurable.

**Comment by lshmouse** [comment]
Thanks for @hit-lacus 's review. I will update the pr asap.

**Comment by lshmouse** [comment]
Format the code and compress the commits. Please help review the pr one more time, thanks~
@hit-lacus

**Pull request merged**